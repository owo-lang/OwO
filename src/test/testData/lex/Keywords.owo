module A where

open import KKSK

a = do
  b <- c <|> do
    ala
    xa
  d
  e

a : B -> C
a (D k) case f k of
  | (E q) case g q of
    | (F b) = bla
  | (L a)   = ora
(D k) (P q) = rua

x = cocase
  .(<>) = (++)
  .mempty = []
