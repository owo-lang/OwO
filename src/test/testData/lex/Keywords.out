ModuleToken (0 1 1) (6 1 7)
IdentifierToken "A" (7 1 8) (8 1 9)
WhereToken (9 1 10) (14 1 15)
BraceLToken (16 3 1) (16 3 1)
OpenToken (16 3 1) (20 3 5)
ImportToken (21 3 6) (27 3 12)
IdentifierToken "KKSK" (28 3 13) (32 3 17)
SemicolonToken (34 5 1) (34 5 1)
IdentifierToken "a" (34 5 1) (35 5 2)
EqualToken (36 5 3) (37 5 4)
DoToken (38 5 5) (40 5 7)
BraceLToken (43 6 3) (43 6 3)
IdentifierToken "b" (43 6 3) (44 6 4)
LeftArrowToken (45 6 5) (47 6 7)
IdentifierToken "c" (48 6 8) (49 6 9)
OperatorToken "<|>" (50 6 10) (53 6 13)
DoToken (54 6 14) (56 6 16)
BraceLToken (61 7 5) (61 7 5)
IdentifierToken "ala" (61 7 5) (64 7 8)
SemicolonToken (69 8 5) (69 8 5)
IdentifierToken "xa" (69 8 5) (71 8 7)
BraceRToken (74 9 3) (74 9 3)
SemicolonToken (74 9 3) (74 9 3)
IdentifierToken "d" (74 9 3) (75 9 4)
SemicolonToken (78 10 3) (78 10 3)
IdentifierToken "e" (78 10 3) (79 10 4)
BraceRToken (81 12 1) (81 12 1)
SemicolonToken (81 12 1) (81 12 1)
IdentifierToken "a" (81 12 1) (82 12 2)
ColonToken (83 12 3) (84 12 4)
IdentifierToken "B" (85 12 5) (86 12 6)
RightArrowToken (87 12 7) (89 12 9)
IdentifierToken "C" (90 12 10) (91 12 11)
SemicolonToken (92 13 1) (92 13 1)
IdentifierToken "a" (92 13 1) (93 13 2)
ParenthesisLToken (94 13 3) (95 13 4)
IdentifierToken "D" (95 13 4) (96 13 5)
IdentifierToken "k" (97 13 6) (98 13 7)
ParenthesisRToken (98 13 7) (99 13 8)
CaseToken (100 13 9) (104 13 13)
IdentifierToken "f" (105 13 14) (106 13 15)
IdentifierToken "k" (107 13 16) (108 13 17)
OfToken (109 13 18) (111 13 20)
BraceLToken (114 14 3) (114 14 3)
SeparatorToken (114 14 3) (115 14 4)
ParenthesisLToken (116 14 5) (117 14 6)
IdentifierToken "E" (117 14 6) (118 14 7)
IdentifierToken "q" (119 14 8) (120 14 9)
ParenthesisRToken (120 14 9) (121 14 10)
CaseToken (122 14 11) (126 14 15)
IdentifierToken "g" (127 14 16) (128 14 17)
IdentifierToken "q" (129 14 18) (130 14 19)
OfToken (131 14 20) (133 14 22)
BraceLToken (138 15 5) (138 15 5)
SeparatorToken (138 15 5) (139 15 6)
ParenthesisLToken (140 15 7) (141 15 8)
IdentifierToken "F" (141 15 8) (142 15 9)
IdentifierToken "b" (143 15 10) (144 15 11)
ParenthesisRToken (144 15 11) (145 15 12)
EqualToken (146 15 13) (147 15 14)
IdentifierToken "bla" (148 15 15) (151 15 18)
BraceRToken (154 16 3) (154 16 3)
SemicolonToken (154 16 3) (154 16 3)
SeparatorToken (154 16 3) (155 16 4)
ParenthesisLToken (156 16 5) (157 16 6)
IdentifierToken "L" (157 16 6) (158 16 7)
IdentifierToken "a" (159 16 8) (160 16 9)
ParenthesisRToken (160 16 9) (161 16 10)
EqualToken (164 16 13) (165 16 14)
IdentifierToken "ora" (166 16 15) (169 16 18)
BraceRToken (170 17 1) (170 17 1)
SemicolonToken (170 17 1) (170 17 1)
ParenthesisLToken (170 17 1) (171 17 2)
IdentifierToken "D" (171 17 2) (172 17 3)
IdentifierToken "k" (173 17 4) (174 17 5)
ParenthesisRToken (174 17 5) (175 17 6)
ParenthesisLToken (176 17 7) (177 17 8)
IdentifierToken "P" (177 17 8) (178 17 9)
IdentifierToken "q" (179 17 10) (180 17 11)
ParenthesisRToken (180 17 11) (181 17 12)
EqualToken (182 17 13) (183 17 14)
IdentifierToken "rua" (184 17 15) (187 17 18)
SemicolonToken (189 19 1) (189 19 1)
IdentifierToken "x" (189 19 1) (190 19 2)
EqualToken (191 19 3) (192 19 4)
CocaseToken (193 19 5) (199 19 11)
BraceLToken (202 20 3) (202 20 3)
DotToken (202 20 3) (203 20 4)
ParenthesisLToken (203 20 4) (204 20 5)
OperatorToken "<>" (204 20 5) (206 20 7)
ParenthesisRToken (206 20 7) (207 20 8)
EqualToken (208 20 9) (209 20 10)
ParenthesisLToken (210 20 11) (211 20 12)
OperatorToken "++" (211 20 12) (213 20 14)
ParenthesisRToken (213 20 14) (214 20 15)
SemicolonToken (217 21 3) (217 21 3)
DotToken (217 21 3) (218 21 4)
IdentifierToken "mempty" (218 21 4) (224 21 10)
EqualToken (225 21 11) (226 21 12)
BracketLToken (227 21 13) (228 21 14)
BracketRToken (228 21 14) (229 21 15)
BraceRToken (230 22 1) (230 22 1)
SemicolonToken (230 22 1) (230 22 1)
BraceRToken (230 22 1) (230 22 1)
